### PR TITLE
Feature/DGJ-94 Improved file upload support

### PR DESCRIPTION
**Pull request by simensma-fresh**
## Summary
This PR fixes an issue where you get a `401 Unauthorized` message when trying to download any of the uploaded files using the file component and the file upload server. The root cause of this has to do with how the Authorization is performed on the file upload server, (relying on `projects`) which is not part of Formsflow.ai.

The PR also includes a custom File Upload Provider / Component to provide some default values when using the Form Builder, and to fix part of the above issue.

## Changes

**Formio File Upload**
- Included https://github.com/formio/formio in the repo so it can be modified. (formio-upload folder)
- Fixed authorization issue, added logging, added use of `helmet` as seen in this commit: https://github.com/bcgov/digital-journeys/commit/63943bb402c4f7c1b711110ee96f54fb00c65dd6
- Added service to `docker-compose`

**Digital Journeys File Upload Component**
- Added custom File Upload component that provides default values for File pattern, download size, and selects the `digital-journeys` upload provider
- Added custom File Upload Provider (`digital-journeys`) that uploads and downloads files to the file upload server

## Screenshots (if applicable)
image

image

image

### CODE REVIEW

File: forms-flow-web/src/constants/constants.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+
+export const FORMIO_JWT_SECRET =
+  (window._env_ && window._env_.REACT_APP_FORMIO_JWT_SECRET) || process.env.REACT_APP_FORMIO_JWT_SECRET;
+
+export const USER_RESOURCE_FORM_ID =
+  (window._env_ && window._env_.REACT_APP_USER_RESOURCE_FORM_ID) ||
+  process.env.REACT_APP_USER_RESOURCE_FORM_ID;
+
+export const FORMIO_FILE_URL =
```
Comment (simensma-fresh) [inline forms-flow-web/src/constants/constants.js:46]: This was added

File: forms-flow-web/src/formComponents/UploadProvider/index.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+          }
+          else {
+            reject(xhr.response || 'Unable to delete file');
+          }
+        };
+        xhr.send(null);
+      });
+    },
+
+    downloadFile(file) {
```
Comment (simensma-fresh) [inline forms-flow-web/src/formComponents/UploadProvider/index.js:145]: This method (+ names) is mostly what differs from the Formio url provider

File: forms-flow-web/src/formComponents/FileUpload/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+export default class DGJFileUpload extends FileComponent{
+
+  static schema(...extend) {
+    return Field.schema({
+      type: 'file',
+      label: 'Upload File',
+      key: 'file',
+      image: false,
+      privateDownload: true,
+      imageSize: '200',
```
Comment (iman-jamali-fw) [inline forms-flow-web/src/formComponents/FileUpload/index.js:24]: What does "size" refer to here?

**Review by iman-jamali-fw — APPROVED** [review]
Looks great! It's really nice to be able to add auth provider and custom components to the Formio.

**Pull request merged**